services/blockchain-indexer/shared/indexer/blockchainIndex.js

```diff
@@ -315,29 +323,27 @@ const deleteIndexedBlocks = async job => {
 				await transactionsTable.deleteByPrimaryKey(forkedTransactionIDs, dbTrx);
 				Signals.get('deleteTransactions').dispatch({ data: forkedTransactions });
 
-				// TODO: Update events fetching logic
-				// const events = await getEventsByHeight(block.height);
-				// if (events.length) {
-				// // TODO: Verify and enable it once pos:validatorStaked schema is exposed from SDK
-				// // Calculate locked amount change and update in key_value_store table for affected tokens
-				// const tokenIDLockedAmountChangeMap = {};
-				// events.forEach(event => {
-				// 	const { data: eventData } = event;
-				// 	// Initialize map entry with BigInt
-				// 	if ([EVENT_NAME.LOCKED, EVENT_NAME.UNLOCKED].includes(event.name)
-				// 		&& !(eventData.tokenID in tokenIDLockedAmountChangeMap) {
-				// 		tokenIDLockedAmountChangeMap[eventData.tokenID] = BigInt(0);
-				// 	}
-
-				// 	// Negate amount to reverse the affect
-				// 	if (event.name === EVENT_NAME.LOCKED) {
-				// 		tokenIDLockedAmountChangeMap[eventData.tokenID] -= eventData.amount;
-				// 	} else if (event.name === EVENT_NAME.UNLOCKED) {
-				// 		tokenIDLockedAmountChangeMap[eventData.tokenID] += eventData.amount;
-				// 	}
-				// });
-				// await updateTotalLockedAmounts(tokenIDLockedAmountChangeMap, dbTrx);
-				// }
+				const events = await getEventsFromCache(block.height);
```

Comment:
Is it possible to fetch the events based on the block ID instead? 🤔

Reply:
Can be done, but would required refactoring of current access pattern and implementation. Also, could not find sdk endpoint to retrieve events by block id, currently we are using `chain_getEvents` which gets by block height